website/server/controllers/api-v3/tasks.js

```diff
@@ -464,8 +464,6 @@ api.updateTask = {
 
     if (!challenge && task.userId && task.challenge && task.challenge.id) {
       sanitizedObj = Tasks.Task.sanitizeUserChallengeTask(updatedTaskObj);
-    } else if (!group && task.userId && task.group && task.group.id) {
```

Comment:
We still need this condition. You can remove the `!group` check, but we still need to run that satanize check if the tasks is a group task. I see it is slightly confusing because of the `sanitizeUserChallengeTask` name, but basically the function is shared between challenges and groups.

You can add a different check `if ((!challenge && task.isChallengeTask) || task.isGroupTask)` which would be an improvement.

Reply:
endet up using `if (!challenge && task.isChallengeTask || task.group && task.group.id)`